Allow to use native promises instead of RSVP

mydea opened the pull request:
I initially tried to get it to work with using `require('rsvp')` when building with Embroider.
However, I couldn't get it to run. I tried adding `rsvp` as a dedicated dependency, as well as `ember-auto-import`, and importing RSVP in an initializer, but nothing made it available under `require('rsvp')` in my embroider app.

I ended up adding an option that allows to opt out of using the RSVP wrapper at all. I think this is an acceptable and backwards compatible opt-out option that should be OK for apps that target only browsers that do not target IE11 anymore.

Let me know if you think this works, or if a different approach would work better.

Closes https://github.com/ember-cli/ember-fetch/issues/622

rwjblue reviewed (commented):
Seems good to me overall. I'd like to make a follow up after this lands to issue a deprecation when _not_ using `preferNative` and `nativePromise` options.

mydea commented on index.js:
```diff
@@ -221,22 +222,50 @@ module.exports = {
+define('fetch', ['exports'], function(exports) {
+  'use strict';
+  var Ember = originalGlobal.Ember;
+  var Promise = originalGlobal.Promise;`;
```
Ah, good point, will change that.

mydea commented:
I have removed the `var Promise = originalGlobal.Promise` part, and also added a ` NATIVE_PROMISE=true yarn test` step to travis to make sure this is covered.
